src/main/java/seedu/address/model/delivery/DeliveryList.java

```diff
@@ -83,6 +85,70 @@ public ObservableList<Delivery> asUnmodifiableObservableList() {
         return internalUnmodifiableList;
     }
 
+    /**
+     * Reverses the backing list.
+     * <p>
+     * Used when sorting the list by a specified attribute, in descending order.
+     */
+    public void reverseDeliveryList() {
+        Collections.reverse(internalList);
+    }
+
+    /**
+     * Sorts the backing list using the {@code Address} attribute of each delivery, in ascending order.
+     */
+    public void sortByAddress() {
+        internalList.sort(Comparator.comparing(d -> d.getAddress().value));
+    }
+
+    /**
+     * Sorts the backing list using the {@code Cost} attribute of each delivery, in ascending order.
+     */
+    public void sortByCost() {
+        internalList.sort((d1, d2) -> (int) (d1.getCost().asFloat() - d2.getCost().asFloat()));
+    }
```

Comment:
Instead of casting into `int`, perhaps would be better to sort by float to maintain the precision - using other methods such as `Float.compare()`

This is because the use of type casting here might be issues in the case where 0.9 - 1.0 would give -0.1, which gets truncated to 0 when cast to an int, leading to incorrect sorting behavior.

Reply:
Thanks for pointing this out! Will push a fix immediately.